File: Data Structures & Algorithms/two-integer-sum/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums: number[], target: number): number[] {
        const store = new Map()

        for(let i = 0; i < nums.length; i++) {
            if(store.has(nums[i])) {
                return [store.get(nums[i]), i]
            }

            store.set(target - nums[i], i)
        }

        return []
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        freq = {}

        for i in range(len(numbers)):
            if numbers[i] in freq:
                return [freq[numbers[i]] + 1, i + 1]

            freq[target - numbers[i]] = i

        
        
        